src/ini.zig

```diff
@@ -858,20 +893,17 @@ pub fn loadNpmrcFromFile(
     };
     defer _ = bun.sys.close(npmrc_file);
 
-    var npmrc_contents = std.ArrayList(u8).init(allocator);
-    defer npmrc_contents.deinit();
-    switch (bun.sys.File.readToEndWithArrayList(bun.sys.File{ .handle = npmrc_file }, &npmrc_contents)) {
-        .result => {},
-        .err => |err| {
-            // TODO: should this exit(1)?
+    const source = switch (bun.sys.File.toSource(".npmrc", allocator)) {
+        .result => |s| s,
+        .err => |e| {
             Output.prettyErrorln("{}\nwhile reading .npmrc \"{s}\"", .{
-                err,
+                e,
                 ".npmrc",
             });
             Global.exit(1);
         },
-    }
-    const source = bun.logger.Source.initPathString(".npmrc", npmrc_contents.items[0..]);
+    };
+    defer allocator.free(source.contents);
```

Comment:
This will free contents before it's used by logger. Maybe move `printForLogLevel` to the bottom of this function?